build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts

```diff
@@ -0,0 +1,135 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "node:fs/promises";
+
+import * as resolve from "resolve.exports";
+import type { PackageJson } from "type-fest";
+import { BaseCommand } from "../../library/index.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import { readPackageJson, readTsConfig } from "../../library/package.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import type { Node10CompatExportData } from "../../library/packageExports.js";
+import type { CommandLogger } from "../../logging.js";
+
+export default class GenerateNode10EntrypointsCommand extends BaseCommand<
+	typeof GenerateNode10EntrypointsCommand
+> {
+	static readonly description =
+		`Generates node10 type declaration entrypoints for Fluid Framework API levels (/alpha, /beta, /internal etc.) as found in package.json "exports"`;
+
+	public async run(): Promise<void> {
+		const packageJson = await readPackageJson();
+
+		const tsconfig = await readTsConfig();
+
+		let emitDeclarationOnly = false;
+		if (tsconfig.compilerOptions?.emitDeclarationOnly !== undefined) {
+			emitDeclarationOnly = true;
+		}
+
+		const mapNode10CompatExportPathToData = mapExportPathsFromPackage(
+			packageJson,
+			emitDeclarationOnly,
+		);
+
+		if (mapNode10CompatExportPathToData.size === 0) {
+			throw new Error(
+				'There are no API level "exports" requiring Node10 type compatibility generation.',
+			);
+		}
+
+		await generateNode10TypeEntrypoints(mapNode10CompatExportPathToData, this.logger);
+	}
+}
+
+export function mapExportPathsFromPackage(
+	packageJson: PackageJson,
+	emitDeclarationOnly: boolean,
+): Map<string, Node10CompatExportData> {
+	const mapKeyToOutput = new Map<string, Node10CompatExportData>();
+
+	const { exports } = packageJson;
+
+	if (typeof exports !== "object" || exports === null) {
+		throw new Error('no valid "exports" within package properties');
+	}
+
+	if (Array.isArray(exports)) {
+		// eslint-disable-next-line unicorn/prefer-type-error
+		throw new Error(`required entrypoints cannot be generated for "exports" array`);
+	}
```

Comment:
Why is this needed?

Reply:
removed this condition with the change you recommended